mso/resource_mso_schema_template_vrf.go

```diff
@@ -189,7 +189,7 @@ func resourceMSOSchemaTemplateVrfCreate(d *schema.ResourceData, m interface{}) e
 	var ipDataPlaneLearning string
 	if ip_data_plane_learning, ok := d.GetOk("ip_data_plane_learning"); ok {
 		ipDataPlaneLearning = ip_data_plane_learning.(string)
-	}
+	} else {ipDataPlaneLearning = "enabled"}
```

Comment:
Is there a reason why you would like to set a value here instead as setting the "ip_data_plane_learning" as required input in the schema?

Reply:
No personal preference at all. But this value is not really a required input in NDO or APIC, v0.9.0 of the provider did not send this value at all.
I think that if we take that into account, the value should either not be sent (if not set) or sent with the documented default.

The code could be modified to not set the `ipDataPlaneLearning` if not set by the operator instead of a default

Reply:
Something like this? Im not a Go coder so im just going for the basics. This also apply to the preferredGroup, right now is working just because is a True/False statement, and if not present it defaults to False. But if the actual default on the APIC were true, it would be a problem.

Basically, non-required statements should not be sent if not set. Harcoding to default works, but if the behaviour changes on the APIC, it will be a conflict.


```go
schemaTemplateVrfApp := models.NewSchemaTemplateVrf("replace", fmt.Sprintf("/templates/%s/vrfs/%s", templateName, Name), Name, displayName, l3m, vzany, preferredGroup)

if ipDataPlaneLearning != "false" {
    schemaTemplateVrfApp.IpDataPlaneLearning = ipDataPlaneLearning
}
```

Reply:
> 

What version of NDO are your currently running? I see 2.3(1c) (Nexus dashboard) but not your NDO specific version. In versions that I run the UI set this value "enabled" which I would argue is sort of like enforcing a default. Currently in the mso-go-client which is used to create the payload ipDataPlane learning is not set when not provided:

```
if ipDataPlaneLearning != "" {
	VrfMap["ipDataPlaneLearning"] = ipDataPlaneLearning
}
```

So to me it seems that not setting is triggering the error and thus a value should always be set in order to have a successful request. This seems to be documented incorrectly in the api documentation, but would need to be tested. 

I am not sure if the behaviour of preferred_group is intended this way but this indeed behaves similar to default of "false" boolean value because the value is initialised in with var which for bool sets to false.

I am not sure what you mean with the APIC part, since the resource is managing NDO schema state.

In order to apply to all the functions I think a change to the schema would be easiest:

```
"ip_data_plane_learning": &schema.Schema{
	Type:     schema.TypeString,
	Optional: true,
	Default: "enabled",
	ValidateFunc: validation.StringInSlice([]string{
		"disabled",
		"enabled",
	}, false),
},
```

Reply:
Hi. According to the tests i did yestereday, not setting `ipDataPlaneLearning` at all on the API call against NDO, does not triggers an error, but setting it to `""` is what triggers the error. 
Error:
```
{"displayName":"L2_VRF","ipDataPlaneLearning":"","l3MCast":false,"name":"L2_VRF","preferredGroup":false,"vzAnyEnabled":false}}
```
No Error:
```
{"displayName":"L2_VRF","l3MCast":false,"name":"L2_VRF","vzAnyEnabled":false}}
```

However, if understand the code correctly, the GetOk to the ip_data_plane learning should be returning an empty string "", and the mso-go-client should be catching that with the if statement that you mentioned and therefore the ipDataPlaneLearning should not be added to the VrfMap, but for some reason, it is being added even if its an empty string.

Reply:
Found the issue, the fix for the empty string was commited on the mso-go-client 1.12.0 and the terraform module is using the previous version 1.11.2

```
require (
	github.com/ciscoecosystem/mso-go-client v1.11.2
	github.com/hashicorp/go-version v1.6.0 // indirect
	github.com/hashicorp/terraform-plugin-sdk v1.17.1
	github.com/zclconf/go-cty v1.8.1 // indirect
)
```

Seems like @anvitha-jain caught this error and commited a fix on the mso-go-client but the version is not referenced by the terraform module.